### PR TITLE
refactor(flash): integrate cexception into pac193x library

### PR DESCRIPTION
**Pull request by HerrCooker**
Integrate CException into pac193x library

### CODE REVIEW

**Comment by HerrCooker** [comment]
Somehow I broke something. For some reason commits that have nothing to do with this branch, are now listed in it. I will stop working on this, as to not break anything

**Comment by glencoe** [comment]
> Somehow I broke something. For some reason commits that have nothing to do with this branch, are now listed in it. I will stop working on this for now, as to not break anything.
> 
> Important: I forgot to add the breaking change flag to my initial commit. Rebase doesn't work.

So what's the state on this one now? Have you been able to fix the problems you were mentioning? Is this PR ready for review?

**Comment by HerrCooker** [comment]
No I have not been able to fix this yet.

**Comment by glencoe** [comment]
closed because of unclear git history

File: src/sensor/pac193x/include/Pac193x.h [hunk 1]
```diff
@@ -118,10 +106,8 @@ pac193xErrorCode_t pac193xRefreshData(pac193xSensorConfiguration_t sensor);
  *
  * @param sensor[in] configuration of the sensor to use
  * @param info[out]  struct that holds the information read from the sensor
- * @return           returns the error code (0 if everything passed)
  */
-pac193xErrorCode_t pac193xGetSensorInfo(pac193xSensorConfiguration_t sensor,
-                                        pac193xSensorId_t *info);
+void pac193xGetSensorInfo(pac193xSensorConfiguration_t sensor, pac193xSensorId_t *info);
```
Comment (glencoe) [inline src/sensor/pac193x/include/Pac193x.h:110]: turn [out] params into return values

File: src/sensor/pac193x/include/Pac193x.h [hunk 2]
```diff
@@ -135,24 +121,19 @@ pac193xErrorCode_t pac193xGetSensorInfo(pac193xSensorConfiguration_t sensor,
  * @param channel[in]        channel where the measurement should be taken from
  * @param valueToMeasure[in] value to be measured
  * @param value[out]         memory where the retrieved value will be stored
- * @return                   returns the error code (0 if everything passed)
  */
-pac193xErrorCode_t pac193xGetMeasurementForChannel(pac193xSensorConfiguration_t sensor,
-                                                   pac193xChannel_t channel,
-                                                   pac193xValueToMeasure_t valueToMeasure,
-                                                   float *value);
+void pac193xGetMeasurementForChannel(pac193xSensorConfiguration_t sensor, pac193xChannel_t channel,
```
Comment (glencoe) [inline src/sensor/pac193x/include/Pac193x.h:125]: see above

File: src/sensor/pac193x/include/Pac193x.h (hunk 1 above)
Comment (HerrCooker) [reply src/sensor/pac193x/include/Pac193x.h:110]: I checked the functions again. They return the sensor info through the info pointer, which is passed to the function.
So AFAIK there is no return value needed, right?

Comment (HerrCooker) [reply src/sensor/pac193x/include/Pac193x.h:110]: @glencoe

Comment (glencoe) [reply src/sensor/pac193x/include/Pac193x.h:110]: The idea is for the function to return the pac193xSensorId_t by value as in
```c
//...
pac193xSensorId_t info;
get_info(&info);
//... do more stuff
return info;
}

**Pull request closed without merging**